### PR TITLE
User mgmt changes PCM-1445

### PR DESCRIPTION
**Pull request by gandhikeyur**
@renujhamtani @pbathia Please review

### CODE REVIEW

File: uds.js [hunk 1]
```diff
@@ -295,5 +297,62 @@
...
+        if (!$.isFunction(onSuccess)) { throw 'onSuccess callback must be a function'; }
+        if (!$.isFunction(onFailure)) { throw 'onFailure callback must be a function'; }
+        var url = udsHostName.clone().setPath('/user?resourceProjection=Full&where=((parentRoleId is '+parentRoleId+') or (roleName is"'+roleName+'" ))');
+        return executeUdsAjaxCall(onSuccess,onFailure,url,'GET');
+    };
+
+    uds.getRoleParentDetails = function (onSuccess, onFailure,parentUserId,parentRoleId,roleName) {
+        if (!$.isFunction(onSuccess)) { throw 'onSuccess callback must be a function'; }
+        if (!$.isFunction(onFailure)) { throw 'onFailure callback must be a function'; }
+        var url = udsHostName.clone().setPath('/user?resourceProjection=Full&where=((parentRoleId is '+parentRoleId+' and roleParentUserId is "'+parentUserId+'") or (roleName is"'+roleName+'" and roleUserId is "'+parentUserId+'"))');
```
Comment (pbathia) [inline uds.js:326]: Just suggestion, can we use addQueryParam for where, so that we will have consistent coding style.

File: uds.js [hunk 2]
```diff
@@ -295,5 +297,62 @@
...
+        if (!$.isFunction(onFailure)) { throw 'onFailure callback must be a function'; }
+        var url = udsHostName.clone().setPath('/user/metadata/roles');
+        url.addQueryParam('where',encodeURIComponent(query));
+        return executeUdsAjaxCall(onSuccess,onFailure,url,'GET');
+    };
+
+    uds.getRoleDetails = function (onSuccess, onFailure,parentRoleId,roleName) {
+        if (!$.isFunction(onSuccess)) { throw 'onSuccess callback must be a function'; }
+        if (!$.isFunction(onFailure)) { throw 'onFailure callback must be a function'; }
+        var url = udsHostName.clone().setPath('/user?resourceProjection=Full&where=((parentRoleId is '+parentRoleId+') or (roleName is"'+roleName+'" ))');
```
Comment (pbathia) [inline uds.js:319]: If possible please use addQueryParam.

**Comment by renujhamtani** [comment]
I am merging this so that Vikas can work on the review comments and send another PR.

**Comment by renujhamtani** [comment]
@vrathee Please see the comments. I have already merged this. Please take latest and send another PR. After reviewing again, I will tag this.

**Pull request merged**